### PR TITLE
server: log the actual error before return "mysql.ErrBadConn"

### PR DESCRIPTION
**Pull request by zz-jason**

### CODE REVIEW

**Comment by zz-jason** [comment]
@coocood @shenli PTAL

**Comment by coocood** [comment]
LGTM

File: server/packetio.go [hunk 1]
```diff
@@ -140,6 +141,7 @@ func (p *packetIO) writePacket(data []byte) error {
 	data[3] = p.sequence
 
 	if n, err := p.bufWriter.Write(data); err != nil {
+		terror.Log(errors.Trace(err))
```
Comment (tiancaiamao) [inline server/packetio.go]: Is there any kind of error other than 'the remote connect is closed' ?

Comment (tiancaiamao) [reply server/packetio.go]: It should already been logged in error trace information in dispatch().

**Pull request merged**